### PR TITLE
get store featured v.2

### PR DESCRIPTION
**Pull request by NavisGames**
## Description

Get the new thing from Henrik's api  get_store_featured_v2 in the wrapper. changed the endpoint's init, responses and the store_featured endpoint

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/raimannma/ValorantAPI/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/raimannma/ValorantAPI/blob/master/CONTRIBUTING.md) guide.
- [x] I've updated the code style using `make codestyle`.  [i formatted like you did, so nothing changed i hope]
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

**Comment by NavisGames** [comment]
there is still one problem: if its a list like "Bundle[0]" you cant see like items, bundle_uuid etc. and if you do "Bundle" you can see it, but it gives an AttributeError 

image
there should it be (is correct)

image
but its there (gives attribute error)

**Comment by raimannma** [comment]
valo_api/endpoints/store_featured.py

Here you have the wrong typing it's not BundleV2 but List[BundleV2]

File: valo_api/endpoints/store_featured.py [hunk 1]
```diff
@@ -19,18 +20,33 @@ def get_store_featured_v1(**kwargs) -> StoreFeaturedV1:
...
-def get_store_featured(version: str, **kwargs) -> StoreFeaturedV1:
+    Args:
+        **kwargs: Any additional arguments to pass to the endpoint.
+
+    Returns:
+        BundleV2: Bundles featured fetched from the API.
+    """
+    return get_store_featured("v2", **kwargs)
+
+def get_store_featured(version: str, **kwargs) -> Union[StoreFeaturedV1, BundleV2]:
```
Comment (raimannma) [inline valo_api/endpoints/store_featured.py:37]:
```suggestion
def get_store_featured(version: str, **kwargs) -> Union[StoreFeaturedV1, List[BundleV2]]]:
```

File: valo_api/endpoints/store_featured.py [hunk 2]
```diff
@@ -19,18 +20,33 @@ def get_store_featured_v1(**kwargs) -> StoreFeaturedV1:
     """
     return get_store_featured("v1", **kwargs)
 
+def get_store_featured_v2(**kwargs) -> BundleV2:
```
Comment (raimannma) [inline valo_api/endpoints/store_featured.py:23]:
```suggestion
def get_store_featured_v2(**kwargs) -> List[BundleV2]]:
```

File: valo_api/endpoints/store_featured.py [hunk 3]
```diff
@@ -47,5 +63,7 @@ def get_store_featured(version: str, **kwargs) -> StoreFeaturedV1:
         raise ValoAPIException(
             ErrorResponse.from_dict(headers=headers, **response_data)
         )
-
-    return StoreFeaturedV1.from_dict(**response_data["data"])
+    if version == "v1":
+        return StoreFeaturedV1.from_dict(**response_data["data"])
+    elif version == "v2":
```
Comment (raimannma) [inline valo_api/endpoints/store_featured.py:68]:
```suggestion
    else:
```

File: valo_api/endpoints/__init__.py [hunk 4]
```diff
@@ -32,7 +32,7 @@
 )
 from valo_api.endpoints.raw import get_raw_data, get_raw_data_v1
 from valo_api.endpoints.status import get_status, get_status_v1
-from valo_api.endpoints.store_featured import get_store_featured, get_store_featured_v1
+from valo_api.endpoints.store_featured import get_store_featured, get_store_featured_v1, get_store_featured_v2
```
Comment (raimannma) [inline valo_api/endpoints/__init__.py:35]:
```suggestion
from valo_api.endpoints.store_featured import (
    get_store_featured,
    get_store_featured_v1,
    get_store_featured_v2,
)
```

File: valo_api/endpoints/store_featured.py [hunk 5]
```diff
@@ -19,18 +20,33 @@ def get_store_featured_v1(**kwargs) -> StoreFeaturedV1:
...
     Args:
         version: The version of the endpoint to use.
             One of the following:
             v1 (Version 1)
+            v2 (Version 2)
         **kwargs: Any additional arguments to pass to the endpoint.
 
     Returns:
         StoreFeaturedV1: Store featured fetched from the API.
+        BundleV2: Bundles featured fetched from the API.
```
Comment (raimannma) [inline valo_api/endpoints/store_featured.py:49]:
```suggestion
        List[BundleV2]: Bundles featured fetched from the API.
```

File: valo_api/endpoints/store_featured.py [hunk 6]
```diff
@@ -19,18 +20,33 @@ def get_store_featured_v1(**kwargs) -> StoreFeaturedV1:
...
+
+    This is the same as
+    :py:meth:`get_store_featured(version="v2", **kwargs) <get_store_featured>`
 
-def get_store_featured(version: str, **kwargs) -> StoreFeaturedV1:
+    Args:
+        **kwargs: Any additional arguments to pass to the endpoint.
+
+    Returns:
+        BundleV2: Bundles featured fetched from the API.
```
Comment (raimannma) [inline valo_api/endpoints/store_featured.py:33]:
```suggestion
        List[BundleV2]: Bundles featured fetched from the API.
```

**Pull request closed without merging**